__tests__/unit-tests/existingSouceMiddleware.test.js

```diff
@@ -0,0 +1,44 @@
+const existingSourceMiddleware = require('../../src/middlewares/existingSourceMiddleware');
+const falsyValues = require('../../__mocks__/falsyValues')
+const path = require('path');
+
+describe('existingSourceMiddleware', () => {
+    const mockRes = {};
+    const mockNext = jest.fn();
+
+    mockRes.status = jest.fn().mockReturnValue(mockRes);
+    mockRes.sendFile =  jest.fn()
+
+    it.each([
+        [
+            "theGuardian",
+            "theTimes",
+            "theTelegraph",
+            "metroUK",
+            "newsBytes",
+            "bbcNews",
+            "skyNews",
+            "reuters",
+            "cnn",
+            "cnbc",
+            "abcNews",
+            "theNewYorkTimes",
+            "abpNews",
+            "zeeNews",
+            "theEconomicTimes",
+            "theGlobeAndMail",
+            "euroNews"
+        ]
```

Comment:
In your opinion, would it be bad practice to automate this? e.g.

```js
const sources = require('path/to/data.sources.js');
...
...
const sourceNames = [];

	for (let i = 0; i < sources.length; i++) {
		let source = sources[i];
		
		sourceNames[i] = source.name.toLowerCase().replace(/ /g, '');
		}
		
// then give sourceNames instead of the manually written list
	
	it.each([sourceNames])('should pass control to the next middleware when a valid source Id has been sent in the requ...

```

Then each new source added will not require altering this test.

Reply:
IMO I wouldn't do that.   

What if a developer creates an error inside the sources array e.g creates a typo or removes an item by accident?    

The tests will always pass and you're no longer relying on expectations laid out in the tests.  I'd feel more confident if the test governed how the list should look.  Making a small adjustment in the test is a small price to pay.  I'm happy to be challenged though.

Reply:
This is precisely why I had reservations and wanted to check. You're absolutely right, plus now future contributors will have the added benefit of becoming more familiar with testing! :raised_hands: 

I'll merge as it is. Thanks again for these contributions!